// @aws-sdk/client-ivs/dist-es/commands/BatchGetChannelCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { BatchGetChannelRequest, BatchGetChannelResponse } from "../models/models_0";
import { deserializeAws_restJson1BatchGetChannelCommand, serializeAws_restJson1BatchGetChannelCommand, } from "../protocols/Aws_restJson1";
var BatchGetChannelCommand = (function (_super) {
    __extends(BatchGetChannelCommand, _super);
    function BatchGetChannelCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    BatchGetChannelCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IvsClient";
        var commandName = "BatchGetChannelCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: BatchGetChannelRequest.filterSensitiveLog,
            outputFilterSensitiveLog: BatchGetChannelResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    BatchGetChannelCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1BatchGetChannelCommand(input, context);
    };
    BatchGetChannelCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1BatchGetChannelCommand(output, context);
    };
    return BatchGetChannelCommand;
}($Command));
export { BatchGetChannelCommand };
