// @aws-sdk/client-ivs/dist-es/commands/CreateChannelCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateChannelRequest, CreateChannelResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateChannelCommand, serializeAws_restJson1CreateChannelCommand, } from "../protocols/Aws_restJson1";
var CreateChannelCommand = (function (_super) {
    __extends(CreateChannelCommand, _super);
    function CreateChannelCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateChannelCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IvsClient";
        var commandName = "CreateChannelCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateChannelRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateChannelResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateChannelCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateChannelCommand(input, context);
    };
    CreateChannelCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateChannelCommand(output, context);
    };
    return CreateChannelCommand;
}($Command));
export { CreateChannelCommand };
