// @aws-sdk/client-ivs/dist-es/commands/DeleteChannelCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteChannelRequest } from "../models/models_0";
import { deserializeAws_restJson1DeleteChannelCommand, serializeAws_restJson1DeleteChannelCommand, } from "../protocols/Aws_restJson1";
var DeleteChannelCommand = (function (_super) {
    __extends(DeleteChannelCommand, _super);
    function DeleteChannelCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteChannelCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IvsClient";
        var commandName = "DeleteChannelCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteChannelRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteChannelCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteChannelCommand(input, context);
    };
    DeleteChannelCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteChannelCommand(output, context);
    };
    return DeleteChannelCommand;
}($Command));
export { DeleteChannelCommand };
