// @aws-sdk/client-ivs/dist-es/commands/GetStreamCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetStreamRequest, GetStreamResponse } from "../models/models_0";
import { deserializeAws_restJson1GetStreamCommand, serializeAws_restJson1GetStreamCommand, } from "../protocols/Aws_restJson1";
var GetStreamCommand = (function (_super) {
    __extends(GetStreamCommand, _super);
    function GetStreamCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetStreamCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IvsClient";
        var commandName = "GetStreamCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetStreamRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetStreamResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetStreamCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetStreamCommand(input, context);
    };
    GetStreamCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetStreamCommand(output, context);
    };
    return GetStreamCommand;
}($Command));
export { GetStreamCommand };
