// @aws-sdk/client-ivs/dist-es/commands/UpdateChannelCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateChannelRequest, UpdateChannelResponse } from "../models/models_0";
import { deserializeAws_restJson1UpdateChannelCommand, serializeAws_restJson1UpdateChannelCommand, } from "../protocols/Aws_restJson1";
var UpdateChannelCommand = (function (_super) {
    __extends(UpdateChannelCommand, _super);
    function UpdateChannelCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateChannelCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "IvsClient";
        var commandName = "UpdateChannelCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateChannelRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateChannelResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateChannelCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateChannelCommand(input, context);
    };
    UpdateChannelCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateChannelCommand(output, context);
    };
    return UpdateChannelCommand;
}($Command));
export { UpdateChannelCommand };
